core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -372,5 +372,5 @@ private[spark] object SparkConf {
   /**
    * Return whether the given config is a Spark port config.
    */
-  def isSparkPortConf(name: String): Boolean = name.startsWith("spark.") && name.endsWith(".port")
+  def isSparkPortConf(name: String): Boolean = name.startsWith("spark.") && name.contains(".port")
```

Comment:
I understand that, but if you are changing the semantics of this function then you should update the docs to reflect that. Maybe something like
```
Return true if the given config refers to a port that Spark uses or is related to Spark ports in general.
```